File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/MedicinalProductContraindication.ts
import { CodeableConcept } from "./CodeableConcept";
import { Population } from "./Population";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { BackboneElement } from "./BackboneElement";

export type MedicinalProductContraindicationOtherTherapy = BackboneElement & {
    medication?: Reference | CodeableConcept;
    therapyRelationshipType: CodeableConcept;
};

export type MedicinalProductContraindication = DomainResource & {
    disease?: CodeableConcept;
    subject?: Reference[];
    population?: Population[];
    comorbidity?: CodeableConcept[];
    otherTherapy?: MedicinalProductContraindicationOtherTherapy[];
    diseaseStatus?: CodeableConcept;
    therapeuticIndication?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicinalProductContraindication.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Population
from base import DomainResource
from base import Reference
from base import BackboneElement

@dataclass(kw_only=True)
class MedicinalProductContraindicationOtherTherapy(BackboneElement):
    medication_reference: Optional[Reference] = None
    therapy_relationship_type: CodeableConcept
    medication_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicinalProductContraindication(DomainResource):
    disease: Optional[CodeableConcept] = None
    subject: Optional[List[Reference]] = field(default_factory=list)
    population: Optional[List[Population]] = field(default_factory=list)
    comorbidity: Optional[List[CodeableConcept]] = field(default_factory=list)
    other_therapy: Optional[List[MedicinalProductContraindicationOtherTherapy]] = field(default_factory=list)
    disease_status: Optional[CodeableConcept] = None
    therapeutic_indication: Optional[List[Reference]] = field(default_factory=list)